Sources/MapboxDirectionsCLI/ProcessCommand.swift

```diff
@@ -70,6 +97,22 @@ class ProcessCommand<ResponceType : Codable, OptionsType : DirectionsOptions > :
         }
     }
     
+    private func interpolate(route: Route?) -> [CLLocationCoordinate2D?] {
+        guard let route = route else { return [] }
+        
+        var distanceAway: CLLocationDistance = 0
+        let distance = route.distance/route.expectedTravelTime
+        guard let polyline = route.shape else { return [] }
+        var interpolatedCoordinates = [route.shape?.coordinates.first]
```

Comment:
The optional chaining on this line complicates the code that uses this method, because it now has to account for each individual element being `nil`, even though there should be no reason for a GPX route to contain nonexistent locations. If there’s no shape associated with the route, then `polyline` would be `nil` so this method would’ve already returned anyways.